src/main.rs

```diff
@@ -155,7 +158,18 @@ async fn listen_blocks(
         .or_default_provider()
         .or_else(Region::new("eu-central-1"));
     let shared_config = aws_config::from_env().region(region_provider).load().await;
-    let client = Client::new(&shared_config);
+    let mut s3_conf = aws_sdk_s3::config::Builder::from(&shared_config);
+    if !endpoint.is_empty() {
```

Comment:
Add a comment before this line explaining the logic behind `endpoint` what kind of endpoint and why it is necessary, please.